src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -99,9 +124,26 @@ void CableView::mouseUp (const MouseEvent& e)
 
 void CableView::timerCallback()
 {
```

Comment:
I figured instead of calling `repaint()` every time in the `timerCallback()`, it makes more sense to check if anything needs repainting in timerCallback, then repaint() accordingly. Unfortunately because of how `repaint()` just flags bounds to be repainted instead of immediately going to the function, it seems safest to recheck the repaint() scenario a second time once we actually get to the `paint()` function. Like how `MouseOverClickablePort()` is checked in the `timerCallback` and again in the `paint()` function. Checking twice like this feels inefficient to me but also still eliminates a lot of redundant `repaints()`.

It's very possible there is another way to solve this problem so let me know if you have any ideas!

Reply:
Yeah, checking all the ports multiple times isn't great, since we have to do things internally like `getNearestPort()` which has to do a good bit of math. Couldn't we use member variables to cache the values returned by `mouseOverClickablePort()` and company from the `timerCallback()` so we still have those values ready to go in the `paint()` callback?

Reply:
Yeah I was wondering about that too. Could there be a delay between when `repaint()` is called and when `paint()` is actually run? So by the time `paint()` is run the member variable value could have changed? That was my worry with the member variable, but that's kinda the limits of my `paint()` `repaint()` knowledge so maybe thats not a big worry. I'll test out the member variable and see how it works!

Reply:
Well, usually there's not too much delay, but even if there is a delay, the only difference would be that we're painting with a more up-to-date value of the variable, so I think it should be fine.